databuilder/query_language.py

```diff
@@ -285,6 +305,10 @@ def parse_date_if_str(value):
 
 
 class DateFunctions(ComparableFunctions):
+    @staticmethod
+    def _cast(value):
+        return parse_date_if_str(value)
```

Comment:
could the contents of parse_date_if_str be inlined here?

Reply:
I wondered about that too (it would certainly be neater) but there's still one place we need to call it. (And we can't use `other._cast()` here because we can't guarantee that `other` is a `BaseSeries`.) I thought about invoking it as:
```py
other = DateFunctions._cast(other)
```
Which would work correctly, but feels like a hack to me.

Reply:
> we can't guarantee that other is a BaseSeries

I hadn't considered that. Good point. I reckon it's good to merge as is